component-lib/package.json

```diff
@@ -25,8 +25,7 @@
   },
   "homepage": "https://github.com/streamlit/streamlit#readme",
   "dependencies": {
-    "apache-arrow": "^0.17.0",
-    "event-target-shim": "^5.0.1",
+    "apache-arrow": "^11.0.0",
```

Comment:
Will existing components be able to upgrade to this new version of component-lib without much effort? (Will any of these dependency upgrades cause issues?)

Reply:
There are a few problems, but I don't think there are major issues:
* The user needs to update the Typescript as the new version of `apache-arrow` requires a new version of the Typescript, but that's our main goal here. Older versions of typescript have problems with the newer version of create-react-script.
* We dropped support for old browser by dropping event-target-shim, but 
kasim.inan confirmed that everything still works on supported browsers.
* We updated the apache-arrow library and this library had some changes to the API and we return this object so if the user relies on it it may require a code update e.g. apache arrow return BigNumber object instead of a BN object.